lib/spack/spack/build_systems/cmake.py

```diff
@@ -73,6 +74,11 @@ class CMakePackage(PackageBase):
 
     build_time_test_callbacks = ['check']
 
+    #: List of names of packages which ``dependent_cmake_args()`` are ignored.
+    #: Everything is ignored if the value is ``all``. Nothing is ignored
+    #: if the value is ``None``.
+    ignore_cmake_args_from = None
```

Comment:
i would prefer to keep it to `all` by default and enable it on per-package basis for those packages which can not find properly their dependencies. Better yet is to switch to a positive logic, i.e. `use_extra_cmake_args_from=` with `None` being default.

Otherwise, all `cmake` package which use `boost` get extra `-DBOOST_ROOT=` set which may not be right for each and every dependent package.

So in order to keep Spack more stable, let's not modify the default behaviour unless the package is known to have issues.